### ISSUE TITLE
How to get score?

### ISSUE THREAD

**Issue by zhouchch3** [ellisdg/3DUnetCNN#294]
Please, anyone can tell me how to get the score of prediction? like accuracy of classification. Thank you

**Comment by ellisdg** [comment]
1. Yes, the dice score should be consistent with the reported dice scores. However, those dice scores were computed on the validation set by the BRATS challenge organizers. I do not have access to their validation dataset.
2. I have not calculated any IoU scores for my experiments.
3. The way my framework operates is that it requires a "Sequence" or "Loader" that reads the input and target data from file and then passes it to the model for 3D data. The sequences that I have made read in NIFTI files which are designed for medical imaging. I'm not familiar with ShapeNet, but to get it to work you would need to either convert the data set to NIFTI format or create a custom sequence to load the data.

**Comment by zhouchch3** [comment]
Thank you! I have another question: Why is the number of channels different between images and target? To verify this, I print the shapes in _epoch_training_ function:
`    for i, (images, target) in enumerate(train_loader):
        print("images.shape: ", images.shape)
        print("target.shape: ", target.shape)
`
output:
`
images.shape: [1, **4**, 112, 112, 112]
target.shape: [1, **3**, 112, 112, 112]
`
As the output is shown, the image has 4 channels, but the target has 3 channels. I thought that both image and target have 3 channels meaning RGB. Could you explain why the image has an extra channel?
Thanks for your kindness.
@ellisdg

**Comment by ellisdg** [comment]
I assume you are referring to the BRATS dataset and model.
The input channels refer to separate MR acquisition parameters. They are T1 weighted (T1w), T1w with contrast enhancing agent, FLAIR, and T2 weighted.
The output channels refer to separate labeled regions: whole tumor, enhancing tumor, and necrotic core.

**Comment by zhouchch3** [comment]
> 

Thank you! Do output channel [0], output channel [1], and output channel [2] represent WT, TC, and ET, respectively?  Because I get the dice score decreasing from channel [0] to channel [2].

**Comment by zhouchch3** [comment]
> 

Thank you! Do output channel [0], output channel [1], and output channel [2] represent WT, TC, and ET, respectively?  Because I get the score of dice decreasing from channel [0] to channel [2].

> I assume you are referring to the BRATS dataset and model. The input channels refer to separate MR acquisition parameters. They are T1 weighted (T1w), T1w with contrast enhancing agent, FLAIR, and T2 weighted. The output channels refer to separate labeled regions: whole tumor, enhancing tumor, and necrotic core.

**Comment by ellisdg** [comment]
> Do output channel [0], output channel [1], and output channel [2] represent WT, TC, and ET, respectively?

Yes, that is correct.
